### PR TITLE
Setup mongodb connection

### PR DESCRIPTION
**Pull request by ryanmt95**
- set up mongodb connection 
- implemented questions Controller 
- the endpoint "/question" is used to get a random question

### CODE REVIEW

File: gibberish_server/app.js [hunk 1]
```diff
@@ -3,12 +3,27 @@ const logger = require('morgan');
...
 const PORT = 4000;
 
-server.listen(PORT)
+// connect to mongo database
+mongoose.connect(process.env.MONGODB_URL, { useUnifiedTopology: true, useNewUrlParser: true, useCreateIndex: true });
+
+const connection = mongoose.connection;
+
+connection.once("open", function () {
+    console.log("MongoDB database connection established successfully");
```
Comment (CiciliaHelena) [inline gibberish_server/app.js:21]: we have imported morgan logger right? I think its better to use that one instead

File: gibberish_server/app.js [hunk 2]
```diff
@@ -3,12 +3,27 @@ const logger = require('morgan');
...
+});
+
+server.listen(PORT, () => {
+    console.log(`Server running on port ${PORT}`);
```
Comment (CiciliaHelena) [inline gibberish_server/app.js:25]: same here

File: gibberish_server/controllers/questions_controller.js [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+const QuestionBank = require('../mongo_models/question_bank');
+const questions = require('../sample_questions')
+
+class questionsController {
+
+    // populate question bank collection in mongodb 
+    static createQuestions(req, res) {
+        for (let question of questions) {
+            QuestionBank.create(question, (err) => {
+                console.log(err)
```
Comment (CiciliaHelena) [inline gibberish_server/controllers/questions_controller.js:10]: here

File: gibberish_server/mongo_models/question_bank.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            unique: true,
+            required: true
+        }
+    },
+    { collection: 'questionbank' }
+);
+
+const QuestionBank = mongoose.model('QuestionBank', questionBankSchema)
+
+module.exports = QuestionBank
```
Comment (CiciliaHelena) [inline gibberish_server/mongo_models/question_bank.js:23]: new line

File: gibberish_server/controllers/questions_controller.js [hunk 5]
```diff
@@ -1,3 +1,31 @@
-questionController() {
-    return true;
+const QuestionBank = require('../mongo_models/question_bank');
+const questions = require('../sample_questions')
+
+class questionsController {
```
Comment (CiciliaHelena) [inline gibberish_server/controllers/questions_controller.js:4]: class name should start with upper case

File: gibberish_server/app.js (hunk 1 above)
Comment (ryanmt95) [reply gibberish_server/app.js:21]: I don't think morgan logger is used as a replacement for console.log. It basically logs the requests made by the client to our server. https://coralogix.com/log-analytics-blog/morgan-npm-logger-the-complete-guide/

Comment (CiciliaHelena) [reply gibberish_server/app.js:21]: oh so it is only used to log the req and res?

**Pull request merged**